File: JINSEOP/프로그래머스_WEEK14/문자열 내 p와 y의 개수.cpp
#include <string>
#include <iostream>
using namespace std;

pair<int, int> comparer;
void compare_p_y(string s){
    int p_cnt = 0, y_cnt = 0;

    for(char c : s){
        if(c == 'p' || c == 'P')
            p_cnt++;
        if(c == 'y' || c == 'Y')
            y_cnt++;
    }
    
    comparer.first = p_cnt;
    comparer.second = y_cnt;
}

bool solution(string s)
{
    bool answer = true;
    
    compare_p_y(s);
    
    if(comparer.first == comparer.second)
        answer = true;
    else answer = false;

    return answer;
}

File: DONGUK/프로그래머스_WEEK_14/문자열 내 p와 y의 개수.py
def solution(s):
    s = s.lower()
    p_cnt, y_cnt = 0, 0
    
    for i in range(len(s)):
        if s[i] == 'p':
            p_cnt += 1
        elif s[i] == 'y':
            y_cnt += 1
    
    if p_cnt == y_cnt:
        return True
    else:
        return False